src/proj.rs

```diff
@@ -892,7 +889,6 @@ impl Drop for Proj {
                 proj_area_destroy(area)
             }
             proj_destroy(self.c_proj);
```

Comment:
There is a change in ordering here, right?

previously:
```
proj_destroy(self.c_proj);
proj_context_destroy(self.ctx);
proj_cleanup();
```

Now we're effectively calling:
```diff
proj_destroy(self.c_proj);
-proj_context_destroy(self.ctx);
proj_cleanup();
+proj_context_destroy(self.ctx); // via Drop impl on ThreadContext
```

Which seems at odds with the comment: 
> // NB do NOT call until proj_destroy and proj_context_destroy have both returned:

I don't actually know the repercussions of this though.

Reply:
Wow wow great job catching this 😮

The only way I can think of to work around this is with https://doc.rust-lang.org/std/mem/struct.ManuallyDrop.html is clunky. Ugh

Reply:
Also note to future selves: think about a way to write a regression test ensuring we call these FFI calls in the right order on drop

Reply:
Would it make sense to move `proj_cleanup` into `ThreadContext#drop`?

Reply:
Maybe! Will need to think on it. Also relevant: https://github.com/georust/proj/issues/75